railties/test/commands/server_test.rb

```diff
@@ -22,6 +22,22 @@ def test_environment_without_server_option
     assert_nil options[:server]
   end
 
+  def test_handler_explicit_option
+    args = ["-x", "thin"]
+    options = parse_arguments(args)
+    assert_equal "thin", options[:server]
+  end
+
+  def test_handler_mistype
+    run_command(["-x", "tin"])
+    assert_match(/Could not find handler 'tin'. Maybe you meant 'thin' or 'cgi'/, output)
+  end
+
+  def test_handler_mistype_in_positional_argument
+    run_command(["tink"])
+    assert_match(/Could not find handler 'tink'. Maybe you meant 'thin' or 'cgi'/, output)
```

Comment:
Don't think `attr_reader :output` and `@output` makes sense. We can just do:

```ruby
assert_match(/Could not find handler 'tink'. Maybe you meant 'thin' or 'cgi'/, run_command(["tink"]))
```

Then it'll match our test/application/test_runner_test.rb tests too.